strings/base_coroutine_foundation.h

```diff
@@ -76,6 +76,35 @@ namespace winrt::impl
         return async.GetResults();
     }
 
+    struct disconnect_aware_handler
+    {
+        disconnect_aware_handler(std::experimental::coroutine_handle<> handle)
+            : m_handle(handle) { }
+
+        disconnect_aware_handler(disconnect_aware_handler&& other)
+            : m_context(std::move(other.m_context))
+            , m_handle(std::exchange(other.m_handle, {})) { }
+
+        ~disconnect_aware_handler()
+        {
+            if (m_handle) Complete();
+        }
+
+        void operator()(Windows::Foundation::IAsyncInfo const&, Windows::Foundation::AsyncStatus)
+        {
+            Complete();
+        }
+
+    private:
+        std::experimental::coroutine_handle<> m_handle;
+        impl::com_ref<impl::IContextCallback> m_context = apartment_context();
```

Comment:
This can simply be `com_ptr`. No need for `impl::com_ref` unless you're not sure what kind of thing `T` is.